packages/swayze/lib/src/widgets/headers/gestures/header_gesture_detector.dart

```diff
@@ -250,67 +273,231 @@ class _HeaderGestureDetectorState extends State<HeaderGestureDetector> {
     );
   }
 
+  /// Handles drag starts that should start dragging a header around.
+  void handleStartDraggingHeader(
+    DragStartDetails details,
+    Range selectionRange,
+  ) {
+    setCursorState(SystemMouseCursors.basic);
+    Actions.invoke(
+      context,
+      HeaderDragStartIntent(
+        draggingPosition: details.localPosition,
+        headers: selectionRange,
+        axis: widget.axis,
+      ),
+    );
+  }
+
+  /// Handles header dragging updates.
+  void handleUpdateDraggingHeader(
+    DragUpdateDetails gestureDetails,
+    _HeaderGestureDetails details,
+  ) {
+    Actions.invoke(
+      context,
+      HeaderDragUpdateIntent(
+        draggingPosition: gestureDetails.localPosition,
+        header: details.headerPosition,
+        axis: widget.axis,
+      ),
+    );
+  }
+
+  void handleDragEnd(SwayzeHeaderDragState state) {
```

Comment:
Don't we need to reset the cursor state when drag ends/is canceled?

Reply:
Not really, during the drag we use the standard cursor, so no change is needed.